### PR TITLE
#174 bug fix

### PR DESCRIPTION
**Pull request by dusekj4**
# Description
- Fixed problem that enabled user to be in mupltiple lobbys
- Add loader when joining lobby, while testing i found out that if servers are overloaded you just watch nothing happening on join button click, not sure if all fail events are handled, this needs check

# Test

- bug was found when you join the lobby and then switch to custom multiplayer and picked other lobby
- try to recreate beeing in multiple rooms

### CODE REVIEW

File: src/renderer/views/multiplayer/custom.vue [hunk 1]
```diff
@@ -226,6 +240,10 @@ async function attemptJoinBattle(battle: SpadsBattle) {
     if (battle.battleOptions.passworded) {
         passwordPromptOpen.value = true;
     } else {
+        loading.value = true;
+        // if user is in a other lobby, leave it
+        api.session.onlineBattle.value && (await api.comms.request("c.lobby.leave"));
```
Comment (Jazcash) [inline src/renderer/views/multiplayer/custom.vue:245]: I'm not a huge fan of inline conditional expressions like this, would prefer:
```typescript
if (api.session.onlineBattle.value) {
    await api.comms.request("c.lobby.leave");
}
```

File: src/renderer/views/multiplayer/custom.vue [hunk 2]
```diff
@@ -179,7 +182,18 @@ let active = true;
 
 const abortController = new AbortController();
 
+const watchForLobbyJoin = api.comms.onResponse("s.lobby.updated_client_battlestatus").add(() => {
+    loading.value = false;
+});
+
+const watchForLobbyJoinFailure = api.comms.onResponse("s.lobby.join").add((data) => {
+    data.result === "failure" || data.reason === "Battle locked" ? (loading.value = false) : null;
```
Comment (Jazcash) [inline src/renderer/views/multiplayer/custom.vue:190]: Same as other comment, would prefer this to be an `if` block as it's more readable:
```typescript
if (data.result === "failure" || data.reason === "Battle locked") {
    loading.value = false;
}
```

File: src/renderer/views/multiplayer/custom.vue [hunk 3]
```diff
@@ -4,7 +4,8 @@
 
 <template>
     <div class="flex-row flex-grow gap-md hide-overflow">
-        <div class="flex-col flex-grow gap-md">
+        <Loader v-if="loading"></Loader>
+        <div v-if="!loading" class="flex-col flex-grow gap-md">
```
Comment (Jazcash) [inline src/renderer/views/multiplayer/custom.vue:8]: You can replace `v-if="!loading"` here with `v-else`

**Review by Jazcash — CHANGES_REQUESTED** [review]

File: src/renderer/views/multiplayer/custom.vue (hunk 1 above)
Comment (dusekj4) [reply src/renderer/views/multiplayer/custom.vue]: changed

File: src/renderer/views/multiplayer/custom.vue (hunk 2 above)
Comment (dusekj4) [reply src/renderer/views/multiplayer/custom.vue]: changed

File: src/renderer/views/multiplayer/custom.vue (hunk 3 above)
Comment (dusekj4) [reply src/renderer/views/multiplayer/custom.vue]: changed

**Review by Jazcash — APPROVED** [review]

**Pull request merged**